### PR TITLE
Lint: Tell users that backslashes are no longer necessary after => & |

### LINKED ISSUE #6456: Lint: `=> \` Deprecated at Cylc 6.
### New rules proposal.

The backslash in `=> \`  hasn't been required but there was no upgrade or obsolete message because it's inside the graph. Lint should probably point this out.

Additionally, consider a second almost identical rule for `& \` and `| \` both of which are unecessary at Cylc 8. I'm in favour because;

- Users _should_ be upgrading from Cylc 7 and back and forward compatible workflows should be becoming less common.
- Users can always use `pyproject.toml` to disable the rule.

### PR DESCRIPTION
**Pull request by wxtim**
Closes #6456 

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] Changelog entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] This is a feature, but a lint feature, so suitable to be added to a maintainance release.

### CODE REVIEW

File: cylc/flow/scripts/lint.py [hunk 1]
```diff
@@ -715,6 +721,12 @@ def list_wrapper(line: str, check: Callable) -> Optional[Dict[str, str]]:
         FUNCTION: functools.partial(
             list_wrapper, check=CHECK_FOR_OLD_VARS.findall),
     },
+    'U017': {
+        'short': (
+            '`&` and `|` are line continuations without `\\`'
+        ),
+        FUNCTION: re.compile(r'[&|]\s*\\').findall
+    },
```
Comment (MetRonnie) [inline cylc/flow/scripts/lint.py:729]: I think this should be part of S014

File: changes.d/6459.feat.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Cylc lint now checks for unecessary continuation characters in graph section.
```
Comment (MetRonnie) [inline changes.d/6459.feat.md:1]:
```suggestion
`cylc lint` now checks for unecessary continuation characters in graph section.
```

**Review by markgrahamdawson — APPROVED** [review]

File: cylc/flow/scripts/lint.py (hunk 1 above)
Comment (hjoliver) [reply cylc/flow/scripts/lint.py:729]: "more dead" 😁

File: changes.d/6459.feat.md [hunk 3]
```diff
@@ -0,0 +1 @@
+`cylc lint` now checks for unecessary continuation characters in graph section.
```
Comment (hjoliver) [inline changes.d/6459.feat.md:1]:
```suggestion
`cylc lint` now checks for unnecessary continuation characters in the graph section.
```

File: cylc/flow/scripts/lint.py [hunk 4]
```diff
@@ -547,7 +547,13 @@ def list_wrapper(line: str, check: Callable) -> Optional[Dict[str, str]]:
     'S013': {
         'short': 'Items should be indented in 4 space blocks.',
         FUNCTION: check_indentation
-    }
+    },
+    'S014': {
+        'short': (
+            '`=>` is a line continuation without `\\`'
```
Comment (hjoliver) [inline cylc/flow/scripts/lint.py:553]:
```suggestion
            '`=>` implies line continuation without `\\`'
```

File: cylc/flow/scripts/lint.py [hunk 5]
```diff
@@ -715,6 +721,12 @@ def list_wrapper(line: str, check: Callable) -> Optional[Dict[str, str]]:
         FUNCTION: functools.partial(
             list_wrapper, check=CHECK_FOR_OLD_VARS.findall),
     },
+    'U017': {
+        'short': (
+            '`&` and `|` are line continuations without `\\`'
```
Comment (hjoliver) [inline cylc/flow/scripts/lint.py:726]:
```suggestion
            '`&` and `|` imply line continuation without `\\`'
```

**Review by hjoliver — CHANGES_REQUESTED** [review]
One typo detected and a couple of small tweaks suggested.

(It may be a bit picky, but I'd say "=>" *imply* line continuation rather than being line continuation).

**Review by hjoliver — APPROVED** [review]

**Comment by MetRonnie** [comment]
This has added a new lint rule S014 on 8.3.x, however S014 is already a different rule on master (https://github.com/cylc/cylc-flow/pull/6137), blocking sync PRs for the time being (https://github.com/cylc/cylc-flow/pull/6489). I think this should be reverted and re-opened against master.

**Pull request merged**